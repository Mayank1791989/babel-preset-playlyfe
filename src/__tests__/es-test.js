/* @flow */
import { testParseCode, testExecCode } from './test-utils';

const opts = [null, undefined, { ie10: true }, { ie10: false }];

testParseCode('Allow trailing function commas', {
  opts,
  throws: false,
  code: `
    function Test(
      arg1,
      arg2,
      arg3,
    ) {
      console.log('arg1')
    }
  `,
});

testParseCode('Allow class properties', {
  opts,
  throws: false,
  code: `
    class Test {
      static x = 5;
      y = 10;
    }
  `,
});

testParseCode('Allow Object rest spread', {
  opts,
  throws: false,
  code: `
    const a = {
      name: 'babel',
    };
    const b = {
      core: 'core',
    };
    const c = {
      ...a,
      ...b,
    };
  `,
});

testParseCode('Allow exports of form export name from module', {
  opts,
  throws: false,
  code: `
    export test from 'test-module';
  `,
});

testParseCode('Allow exports of form export * as name from module', {
  opts,
  throws: false,
  code: `
    export * as test from 'test-module';
  `,
});

testParseCode('Allow decorator', {
  opts,
  throws: false,
  code: `
    @generateStyles(() => ({}))
    class Test extends React.Component {
      render() {
        return null;
      }
    }
  `,
});

testParseCode('Default Flow Support is enabled', {
  opts,
  throws: false,
  code: `
    function test(test1: string, test2: number) {}
  `,
});

// NOTE: if react is enabled flow can't be disabled
// as react preset has no option to disable flow (see babel-preset-react)
testParseCode('Can disable flow support', {
  opts: [{ flow: false }],
  throws: true,
  code: `
    function test(test1: string, test2: number) {}
  `,
});

testExecCode('decorators should work with static properties', {
  opts,
  code: `
    function someDecorator(Component) {
      return class WrappedComponent {
        render() {
          <Component {...this.props} />
        }
      }
    }

    @someDecorator
    class Test {
      static propTypes = {
        a: 10,
      };
    }

    // Tests
    // NOTE here 'Test' is 'someDecorator(Test)' not class Test
    expect(Test.propTypes).toBeUndefined();
  `,
});
