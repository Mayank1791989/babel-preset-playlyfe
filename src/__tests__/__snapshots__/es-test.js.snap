// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Support disable import conversion 1`] = `
"
import { test } from 'test';
console.log('test');"
`;

exports[`default dont transform transform async await 1`] = `
"\\"use strict\\";

async function test() {
  var a = await test2();
}"
`;

exports[`transform if Async await enabled 1`] = `
"\\"use strict\\";

var _regenerator = require(\\"babel-runtime/regenerator\\");

var _regenerator2 = _interopRequireDefault(_regenerator);

var _asyncToGenerator2 = require(\\"babel-runtime/helpers/asyncToGenerator\\");

var _asyncToGenerator3 = _interopRequireDefault(_asyncToGenerator2);

var test = function () {
  var _ref = (0, _asyncToGenerator3.default)(_regenerator2.default.mark(function _callee() {
    var a;
    return _regenerator2.default.wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            _context.next = 2;
            return test2();

          case 2:
            a = _context.sent;

          case 3:
          case \\"end\\":
            return _context.stop();
        }
      }
    }, _callee, this);
  }));

  return function test() {
    return _ref.apply(this, arguments);
  };
}();

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }"
`;
